contracts/RMRK/extension/tokenProperties/RMRKTokenPropertiesRepository.sol

```diff
@@ -0,0 +1,471 @@
+    mapping(address => mapping(uint256 => string)) private _stringIdToValue;
+    mapping(address => mapping(uint256 => mapping(uint256 => uint256)))
+        private _stringValueIds;
+
+    mapping(address => mapping(uint256 => mapping(uint256 => address)))
+        private _addressValues;
+    mapping(address => mapping(uint256 => mapping(uint256 => bytes)))
+        private _bytesValues;
+    mapping(address => mapping(uint256 => mapping(uint256 => uint256)))
+        private _uintValues;
+    mapping(address => mapping(uint256 => mapping(uint256 => bool)))
+        private _boolValues;
+
+    struct IssuerSetting {
+        bool registered;
+        bool useOwnable;
+        address issuer;
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function registerAccessControl(
+        address collection,
+        address issuer,
+        bool useOwnable
+    ) external onlyUnregisteredCollection(collection) {
+        (bool ownableSuccess, bytes memory ownableReturn) = collection.call(
+            abi.encodeWithSignature("owner()")
+        );
+
+        if (address(uint160(uint256(bytes32(ownableReturn)))) == address(0)) {
+            revert RMRKOwnableNotImplemented();
+        }
+        if (
+            ownableSuccess &&
+            address(uint160(uint256(bytes32(ownableReturn)))) != msg.sender
+        ) {
+            revert RMRKNotCollectionIssuer();
+        }
+
+        IssuerSetting storage issuerSetting = _issuerSettings[collection];
+        issuerSetting.registered = true;
+        issuerSetting.issuer = issuer;
+        issuerSetting.useOwnable = useOwnable;
+
+        emit AccessControlRegistration(
+            collection,
+            issuer,
+            msg.sender,
+            useOwnable
+        );
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function manageAccessControl(
+        address collection,
+        string memory key,
+        AccessType accessType,
+        address specificAddress
+    ) external onlyRegisteredCollection(collection) onlyIssuer(collection) {
+        uint256 parameterId = _getIdForKey(collection, key);
+
+        _parameterAccessType[collection][parameterId] = accessType;
+        _parameterSpecificAddress[collection][parameterId] = specificAddress;
+
+        emit AccessControlUpdate(collection, key, accessType, specificAddress);
+    }
+
+    /**
+     * @inheritdoc IRMRKTokenPropertiesRepository
+     */
+    function manageCollaborators(
+        address collection,
+        address[] memory collaboratorAddresses,
+        bool[] memory collaboratorAddressAccess
+    ) external onlyRegisteredCollection(collection) onlyIssuer(collection) {
+        if (collaboratorAddresses.length != collaboratorAddressAccess.length) {
+            revert RMRKCollaboratorArraysNotEqualLength();
+        }
+        for (uint256 i = 0; i < collaboratorAddresses.length; i++) {
```

Comment:
For gas savings, no need to do `i = 0` since 0 is the default. Also, `unchecked { ++i; }` is cheaper for increase the counter